File: code_combat/4_Backwoods_Forest/190-Logical_Circle/logic_circle.py
def logic(value, a, b):
    if value:
        return a
    return b

hero.moveXY(20, 24);
a = hero.findNearestFriend().getSecretA()
b = hero.findNearestFriend().getSecretB()
c = hero.findNearestFriend().getSecretC()

hero.moveXY(30, logic(a and b and c, 33, 15))
hero.moveXY(logic(a or b or c, 20, 40), 24)
hero.moveXY(30, logic(a and b and c, 33, 15))

File: projects/code_combat/4_Backwoods_Forest/190-Logical_Circle/logicCircle.js
// Move to the wizard and get their secret values.
hero.moveXY(20, 24);
var secretA = hero.findNearestFriend().getSecretA();
var secretB = hero.findNearestFriend().getSecretB();
var secretC = hero.findNearestFriend().getSecretC();

// If ALL three values are true, take the high path.
// Otherwise, take the low path. Save the 4th value.
var secretD = secretA && secretB && secretC;
if (secretD)
    hero.moveXY(30, 33);
else
    hero.moveXY(30, 15);

// If ANY of the three values are true, take the left path.
// Otherwise, go right. Save the 5th value.
var secretE = secretA || secretB || secretC;
if (secretE)
    hero.moveXY(20, 24);
else
    hero.moveXY(40, 24);

// If ALL five values are true, take the high path.
// Otherwise, take the low path.
if (secretA && secretB && secretC && secretD && secretE)
    hero.moveXY(30, 33);
else
    hero.moveXY(30, 15);
